packages/manager/src/features/Lish/Lish.tsx

```diff
@@ -23,10 +23,78 @@ import type { Tab } from 'src/components/Tabs/TabLinkList';
 
 const AUTH_POLLING_INTERVAL = 2000;
 
+export interface RetryLimiterInterface {
+  retryAllowed: () => boolean
+  reset: () => void
+}
+
+export const RetryLimiter = (maxTries: number, perTimeWindowMs: number): RetryLimiterInterface => {
+  let retryTimes: number[] = [];
+
+  return {
+    retryAllowed: (): boolean => {
+      const now = Date.now();
+      retryTimes.push(now);
+      const cutOffTime = now-perTimeWindowMs;
+      while (retryTimes.length && retryTimes[0] < cutOffTime)
+        retryTimes.shift();
+      return retryTimes.length < maxTries;
+    },
+    reset: (): void => {
+      retryTimes = [];
+    }
+  }
+}
+
+export interface LishErrorInterface {
+  reason: string;
+  grn: string;
+  formatted: string;
+  isExpired: boolean;
+}
+
+export const ParsePotentialLishErrorString = (s: string | null): LishErrorInterface | null => {
+  if (!s) return null;
+
+  let parsed = null;
+  try {
+    parsed = JSON.parse(s);
+  } catch {
+    return null;
+  }
+
+  let grn = (typeof parsed?.grn === "string" ? parsed?.grn : '');
```

Comment:
Since `LishErrorInterface` declares `reason` and `grn` as `string`, we don't need to check that they are strings

```suggestion
  const grn = parsed?.grn ?? '';
```

Can we also address the lint errors?
<img src="https://github.com/user-attachments/assets/14255895-2826-4999-a076-e49448735872" width="600" height="300" />

Reply:
Done!